### DISCUSSION TITLE
Increase tail output (currently 15 lines) on error

### DISCUSSION THREAD

**Discussion by andreineculau** [Homebrew/discussions#1878] [Everyday usage]
Currently when `brew install X` fails, due to a system command, homebrew prints "Last 15 lines from path/to/log.file".
Is it possible to configure that 15 to something bigger? I'd say 3/4 times there's a problem with python (pip to be exact) and all that you see in 15 lines is a stacktrace, rather than an actual error (usually incompatible dependencies).
Locally that's fine, but in the CI, I can only think of two options:
1. increase to print more than 15 lines
2. tee output and, on failure, grep for `^Last 15 lines from ` to get path/to/log.file and do `tail -n30 path/to/log.file`

**Comment by carlocab** [comment]
You can probably configure CI to upload the logs that `brew` generates upon failure. See workflows in Homebrew/core for an example.

Otherwise, you can set an env variable like `HOMEBREW_FAIL_LOG_LINES`. The actual name might be slightly different -- see `man brew` for details.

**Comment by andreineculau** [comment]
Thanks!
I missed the env var... I looked only for HOMEBREW_LOG on that page.

--
http://andreineculau.com
